substrate/client/executor/polkavm/src/lib.rs

```diff
@@ -158,34 +141,34 @@ impl<'r, 'a> FunctionContext for Context<'r, 'a> {
 fn call_host_function(
 	caller: &mut Caller<()>,
 	function: &dyn Function,
-) -> Result<(), polkavm::Trap> {
+) -> Result<(), polkavm::CallError> {
 	let mut args = [Value::I64(0); Reg::ARG_REGS.len()];
 	let mut nth_reg = 0;
 	for (nth_arg, kind) in function.signature().args.iter().enumerate() {
 		match kind {
 			ValueType::I32 => {
-				args[nth_arg] = Value::I32(caller.get_reg(Reg::ARG_REGS[nth_reg]) as i32);
+				args[nth_arg] = Value::I32(caller.instance.reg(Reg::ARG_REGS[nth_reg]) as i32);
 				nth_reg += 1;
 			},
 			ValueType::F32 => {
-				args[nth_arg] = Value::F32(caller.get_reg(Reg::ARG_REGS[nth_reg]));
+				args[nth_arg] = Value::F32(caller.instance.reg(Reg::ARG_REGS[nth_reg]) as u32);
 				nth_reg += 1;
 			},
 			ValueType::I64 => {
-				let value_lo = caller.get_reg(Reg::ARG_REGS[nth_reg]);
+				let value_lo = caller.instance.reg(Reg::ARG_REGS[nth_reg]);
 				nth_reg += 1;
 
-				let value_hi = caller.get_reg(Reg::ARG_REGS[nth_reg]);
+				let value_hi = caller.instance.reg(Reg::ARG_REGS[nth_reg]);
 				nth_reg += 1;
 
 				args[nth_arg] =
 					Value::I64((u64::from(value_lo) | (u64::from(value_hi) << 32)) as i64);
 			},
 			ValueType::F64 => {
-				let value_lo = caller.get_reg(Reg::ARG_REGS[nth_reg]);
+				let value_lo = caller.instance.reg(Reg::ARG_REGS[nth_reg]);
 				nth_reg += 1;
 
-				let value_hi = caller.get_reg(Reg::ARG_REGS[nth_reg]);
+				let value_hi = caller.instance.reg(Reg::ARG_REGS[nth_reg]);
```

Comment:
These are incorrect on 64-bit. Add an `if caller.instance.is_64_bit()` to check if we're on 64-bit; if we're on 32-bit the current code is correct, if we're on 64-bit then the values only use 1 register.

Reply:
So I fixed these based on compilation error and thought that get_reg is the same as reg. How to fix this?

Reply:
What is missing making this code 64bit aware. Registers are always 64bit from point of view of the host. However, the guest will have 32 or 64bit wide registers depending on arch. This means that if `is_64_bit()` it will pass 64bit values in a single registers. Otherwise it will use two (as it is the case in the current code).